code/obj/machinery/espresso.dm

```diff
@@ -155,13 +155,29 @@
 	var/default_carafe = /obj/item/reagent_containers/food/drinks/carafe
 	var/image/fluid_image
 
+	var/emagged = 0
+
 	New()
 		..()
 		UnsubscribeProcess()
 		if (ispath(src.default_carafe))
 			src.my_carafe = new src.default_carafe (src)
 		src.update()
 
+	emag_act(var/mob/user, var/obj/item/card/emag/E)
+
+		if(!src.emagged)
+			if (user)
+				boutput(user, "<span class='notice'>You force the machine to brew something else...</span>")
+
+			src.desc = " It's top of the line NanoTrasen tea technology! Featuring 100% Organic Locally-Grown green leaves!"
+			src.emagged = 1
+			return 1
```

Comment:
Please use TRUE/FALSE defines instead of 1/0 here

Reply:
I'm genuinly unfamiliar with DM, could you elaborate what to do exactly please?